### PR TITLE
Fix state.sls_id not running on ssh minion

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?
`state.sls_id` was running on the master and not the salt-ssh minion. This PR makes sure we are running the state on the salt-ssh minion. I noticed we are calling a lot of the same functions in the state.py wrapper for salt-ssh so I created this funciton: `_ssh_state`. I hope to move the other ones to this but in develop to make sure we don't ruin something in a point release once this is merged forward

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/48318

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by rallytime — APPROVED** [review]
LGTM

**Review by gtmanfred — APPROVED** [review]

**Review by DmitryKuzmenko — APPROVED** [review]

File: salt/client/ssh/wrapper/state.py [hunk 1]
```diff
@@ -32,6 +32,60 @@
 log = logging.getLogger(__name__)
 
 
+def _ssh_state(chunks, __opts__, __context__, __pillar__, __salt__, st_kwargs,
```
Comment (cachedout) [inline salt/client/ssh/wrapper/state.py]: I'm very confused by this. Why are we taking in `__opts__` and `__context__` as arguments for this function when they should already be available in this namespace?

File: salt/client/ssh/wrapper/state.py [hunk 2]
```diff
@@ -32,6 +32,60 @@
...
+    file_refs = salt.client.ssh.state.lowstate_file_refs(
+            chunks,
+            _merge_extra_filerefs(
+                kwargs.get('extra_filerefs', ''),
+                __opts__.get('extra_filerefs', '')
+                )
+            )
+    # Create the tar containing the state pkg and relevant files.
+    trans_tar = salt.client.ssh.state.prep_trans_tar(
+            __opts__,
```
Comment (isbm) [inline salt/client/ssh/wrapper/state.py]: @Ch3LL  This is unused inside the function. I'd propose either remove the option entirely from `prep_trans_tar` or refactor that it is used.

File: salt/client/ssh/wrapper/state.py (hunk 1 above)
Comment (isbm) [reply salt/client/ssh/wrapper/state.py]: And `__pillar__`. Can we fix this, @Ch3LL ?

**Pull request merged**